src/Registration/Registration.tsx

```diff
@@ -14,76 +15,60 @@ type DispatchProps = {
 
 type Props = StateProps & DispatchProps;
 
-type State = {
-  email: string;
-  firstName: string;
-  surname: string;
-  type: UserType | null;
-};
-
-export class _Registration extends Component<Props, State> {
-  public state: State = {
-    email: '',
-    firstName: '',
-    surname: '',
-    type: null,
-  };
-
-  public render() {
-    return (
-      <div>
-        <input
-          value={this.state.firstName}
-          placeholder='firstname'
-          onChange={this.onFirstNameChange}
-        />
-        <input
-          value={this.state.surname}
-          placeholder='surname'
-          onChange={this.onSurnameChange}
-        />
-        <input
-          value={this.state.email}
-          placeholder='email'
-          onChange={this.onEmailChange}
-        />
-        <select value={this.state.type || ''} onChange={this.onTypeChange}>
-          <option value=''>Select...</option>
-          {Object.values(UserType).map(userType => (
-            <option value={userType}>{userType}</option>
-          ))}
-        </select>
-        <button onClick={this.onAddUser}>Submit</button>
-      </div>
-    );
-  }
-
-  private onTypeChange: ChangeEventHandler<HTMLSelectElement> = event => {
-    const value: UserType | null =
-      (event.currentTarget.value as UserType) || null;
-    this.setState({ type: value });
-  };
-
-  private onFirstNameChange: ChangeEventHandler<HTMLInputElement> = event => {
-    this.setState({ firstName: event.currentTarget.value });
-  };
-
-  private onSurnameChange: ChangeEventHandler<HTMLInputElement> = event => {
-    this.setState({ surname: event.currentTarget.value });
-  };
+const _Registration = (props: Props) => {
+  const [firstName, onFirstNameChange] = useField('');
+  const [surname, onSurnameChange] = useField('');
+  const [type, onTypeChange] = useField<UserType | null, HTMLSelectElement>(null);
```

Comment:
This'll break if you change the `type` to something, and then select the "Select..." value. I'll leave it up to you to tell me why.

Reply:
This'll break due to the `type || ''` and the string portion of this union returning as undefined where (user) type can only be of type `UserType || null`. As discussed I'll create a function to take the undefined value and return it as a `null` value.